astropy/io/votable/tree.py

```diff
@@ -2828,7 +2828,7 @@ def _write_binary(self, mode, w, **kwargs):
                     for i, converter in fields_basic:
                         try:
                             chunk = converter(array_row[i], array_mask[i])
-                            assert type(chunk) == type(b'')
```

Comment:
Think this should be kept as is: presumably one doesn't want `chunk` to be a subclass of `Bytes`, so just `type(chunk) is type(b'')`.